File: test/common/function_availability.lua
local M = {};

local function FunctionBinded(func, functionName)
    if functionName == nil then
        FileWriteLineFast("test/output.txt",
         "Function '" .. functionName .. "' are not binded", FM_APPEND);
        return false;
    end;
    return true;
end;

local function FunctionSupported(func, functionName, faFunc)
    if FunctionBinded(func, functionName) then
        local funcAvail = faFunc(functionName);
        if funcAvail == FA_NO then
            FileWriteLineFast("test/output.txt",
             "Function's availability data for function '" .. functionName ..
             "' not found", FM_APPEND);
        elseif funcAvail == FA_DUMMY then
            FileWriteLineFast("test/output.txt",
             "Function '" .. functionName .. "' not supported", FM_APPEND);
        elseif funcAvail == FA_BUGGY then
            FileWriteLineFast("test/output.txt",
             "Function '" .. functionName .. "' are not working correct",
             FM_APPEND);
        elseif funcAvail == FA_PARTIAL then
            FileWriteLineFast("test/output.txt",
             "Function '" .. functionName .. "' partially supported",
             FM_APPEND);
        elseif funcAvail == FA_YES then
            FileWriteLineFast("test/output.txt",
             "Function '" .. functionName .. "' supported", FM_APPEND);
        else
            FileWriteLineFast("test/output.txt",
             "Unexpected function's availability data for function '" ..
             functionName .. "'", FM_APPEND);
        end;
    end;
end;

M.FunctionBinded = FunctionBinded;
M.FunctionSupported = FunctionSupported;

return M;

File: test/common/function_availability.js
var FunctionBinded = function (func, functionName)
{
    if (functionName == null)
    {
        FileWriteLineFast("test/output.txt",
         "Function '" + functionName + "' are not binded", FM_APPEND);
        return false;
    }
    return true;
}

var FunctionSupported = function (func, functionName, faFunc)
{
    if (FunctionBinded(func, functionName))
    {
        var funcAvail = faFunc(functionName);
        switch (funcAvail)
        {
            case FA_NO:
                FileWriteLineFast("test/output.txt",
                 "Function's availability data for function '" +
                 functionName + "' not found", FM_APPEND);
                break;
            case FA_DUMMY:
                FileWriteLineFast("test/output.txt",
                 "Function '" + functionName + "' not supported", FM_APPEND);
                 break;
            case FA_BUGGY:
                FileWriteLineFast("test/output.txt",
                 "Function '" + functionName + "' are not working correct",
                 FM_APPEND);
                break;
            case FA_PARTIAL:
                FileWriteLineFast("test/output.txt",
                 "Function '" + functionName + "' partially supported",
                 FM_APPEND);
                break;
            case FA_YES:
                FileWriteLineFast("test/output.txt",
                 "Function '" + functionName + "' supported", FM_APPEND);
                break;
            default:
                FileWriteLineFast("test/output.txt",
                 "Unexpected function's availability data for function '" +
                 functionName + "'", FM_APPEND);
                break;
        }
    }
}

exports.FunctionSupported = FunctionSupported;
